### PR TITLE
Add Spider Tool

### PR DESCRIPTION
**Pull request by WilliamEspegren**
This PR is inspired by #239, and adds [Spider](https://github.com/spider-rs/spider) search functionality to provide more options for developers when they need agents to use a search engine.

Twitter: [@WilliamEspegren](https://x.com/WilliamEspegren)

### CODE REVIEW

File: phi/tools/spider.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import json
+from spider import Spider as ExternalSpider  # type: ignore
```
Comment (ysolanky) [inline phi/tools/spider.py:2]:
```python
try:
     from spider import Spider as ExternalSpider
except ImportError:
    logger.error("`spider` not installed")
    raise
 ```

Comment (WilliamEspegren) [reply phi/tools/spider.py:2]: Adding it, 2 minutes

Comment (WilliamEspegren) [reply phi/tools/spider.py:9]: Just added it, thanks for the review 🙌

**Comment by WilliamEspegren** [comment]
@ysolanky ready to merge :)

**Review by ashpreetbedi — APPROVED** [review]

**Pull request merged**